scripts/extensions/pool_update.apply

```diff
@@ -44,9 +46,18 @@ def failure_message(code, params):
 
 
 def execute_apply(session, update_package, yum_conf_file):
-    cmd = ['yum', 'upgrade', '-y', '--noplugins', '-c', yum_conf_file, update_package]
     yum_env = os.environ.copy()
     yum_env['LANG'] = 'C'
+
+    cmd = ['yum', 'clean', 'all', '--noplugins', '-c', yum_conf_file]
+    p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, close_fds=True, env=yum_env)
+    output, _ = p.communicate()
```

Comment:
This seems to be duplicated in quite a few places, would `subprocess.check_output` be a suitable replacement?
Or if we need the logging could we have a helper function `call_and_log_cmd`? `remove_group` has almost the same code copy/pasted

Reply:
So I just copied what the existing code was doing for consistency - I agree `check_output` would be cleaner, but I wonder if it's more confusing to have a mix of styles in the same script?

Reply:
I think this is fine for this as it is clear and concise enough.